### PR TITLE
Improvement on the phospho search

### PR DESCRIPTION
**Pull request by oggyfoxy**
Please take a look @oulianov for @Pierre-LouisBJT's url2chat.

### CODE REVIEW

File: asynctest.py [hunk 1]
```diff
@@ -0,0 +1,173 @@
...
+import trafilatura
+from openai import AsyncOpenAI
+from dotenv import load_dotenv
+from typing import List, Tuple, Dict
+import time
+import openai
+from urllib.parse import urlparse
+import config
+import phospho
+from testgoogleapi import get_relevant_urls_from_google
```
Comment (oulianov) [inline asynctest.py:13]: please add this to requirements

File: asynctest.py [hunk 2]
```diff
@@ -0,0 +1,173 @@
...
+
+# Initialize OpenAI client and other libraries
+load_dotenv()
```
Comment (oulianov) [inline asynctest.py:16]: remove this please

File: testgoogleapi.py [hunk 3]
```diff
@@ -0,0 +1,27 @@
+from typing import List
+from googleapiclient.discovery import build
+from dotenv import load_dotenv
+import os
+
+load_dotenv()
+
+
+my_api_key = os.getenv("GOOGLE_API_KEY")
+my_cse_id = os.getenv("GOOGLE_CSE_ID")
```
Comment (oulianov) [inline testgoogleapi.py:10]: How do I get those? Can you send me a link

File: asynctest.py
Comment (oulianov) [inline asynctest.py:1]: Instead of creating a new file, please edit the existing one

**Review by oulianov — CHANGES_REQUESTED** [review]
Good start!

File: testgoogleapi.py (hunk 3 above)
Comment (oggyfoxy) [reply testgoogleapi.py]: I added instructions in the README.md, feel free to reach out if you need

**Pull request merged**